### PR TITLE
Consolidate reqs into top level requirements_dev.txt

### PR DESCRIPTION
**Pull request by mcgibbon**
## Purpose

Currently Pace has many `requirements.txt` files and variants. This makes it difficult to understand how the environment is managed and make changes. This PR consolidates all of them into the top level requirements files.

## Infrastructure changes:

- deleted requirements files in driver examples, in each subpackage top level, and in `fv3core/requirements`
- deleted top level `requirements.txt`, as `requirements.dev` serves its purpose
- deleted `fv3core/.jenkins/install_virtualenv.sh` and modified `fv3core/.jenkins/jenkins.sh` to use the top-level `.jenkins/install_virtualenv.sh`

## Checklist
Before submitting this PR, please make sure:

- [x] You have followed the coding standards guidelines established at [Code Review Checklist](https://drive.google.com/file/d/1R0nqOxfYnzaSdoYdt8yjx5J482ETI2Ft/view?usp=sharing).
- [x] Docstrings and type hints are added to new and updated routines, as appropriate
- [x] All relevant documentation has been updated or added (e.g. README, CONTRIBUTING docs)
- [x] For each public change and fix in `pace-util`, HISTORY has been updated
- [x] Unit tests are added or updated for non-stencil code changes

### CODE REVIEW

**Comment by mcgibbon** [comment]
launch jenkins

**Comment by mcgibbon** [comment]
launch jenkins

File: requirements_dev.txt [hunk 1]
```diff
@@ -1,8 +1,17 @@
...
+pytest-profiling
 pytest-cov
 mpi4py
-dace>=0.14
+xarray
+zarr
+netCDF4
+cftime
+fv3config>=0.9.0
+git+https://github.com/spcl/dace.git@v0.14rc2
```
Comment (FlorianDeconinck) [inline requirements_dev.txt:12]: This is wrong, this needs to be dace 0.14.

**Comment by mcgibbon** [comment]
launch jenkins

**Comment by mcgibbon** [comment]
jenkins passed other than random failures on gpu backend https://jenkins.ginko.ch/job/pace_PR/2763/

**Comment by mcgibbon** [comment]
launch jenkins

**Comment by mcgibbon** [comment]
launch jenkins

**Comment by mcgibbon** [comment]
pace-physics-regression-PR is [failing on the gpu parallel tests with a segfault](https://jenkins.ginko.ch/job/pace-physics-regression_PR/2251/backend=gt_gpu,experiment=c12_6ranks_baroclinic_dycore_microphysics,runaction=run_physics_parallel_regression_tests,slave=daint_submit/console), failed on the last run with another intermittent error.

**Review by FlorianDeconinck — COMMENTED** [review]
LGTM

**Review by FlorianDeconinck — APPROVED** [review]

**Comment by mcgibbon** [comment]
launch jenkins

**Comment by mcgibbon** [comment]
launch jenkins

**Comment by mcgibbon** [comment]
launch jenkins

**Comment by mcgibbon** [comment]
launch jenkins

**Comment by mcgibbon** [comment]
launch jenkins

**Comment by mcgibbon** [comment]
OK, after enough repeated launches of jenkins, I can say this PR reproducibly causes the [gt:gpu physics parallel tests to fail](https://jenkins.ginko.ch/job/pace-physics-regression_PR/2263/backend=gt_gpu,experiment=c12_6ranks_baroclinic_dycore_microphysics,runaction=run_physics_parallel_regression_tests,slave=daint_submit/console), and all non-numpy backends to [fail to compile for fv3core](https://jenkins.ginko.ch/job/pace-fv3core-regression_PR/2553/).

**Pull request merged**